src/Sniffer.php

```diff
@@ -0,0 +1,278 @@
+            /** @var FencedCode $block */
+            foreach ($codeBlocks as $block) {
+                if (strtoupper($block->getInfo()) !== $lintLanguage) {
+                    continue;
+                }
+                // We only want to count relevant code blocks
+                $n++;
+
+                // $startAt is the line in the md file where the ```php line sits
+                $startAt = $block->getStartLine();
+
+                // Pad the content out so we have an accurate line count, and prepend a php code opening tag
+                $content = str_repeat(PHP_EOL, $startAt - 1);
+                $content .= '<?php' . PHP_EOL . $block->getLiteral();
+
+                // Report each block separately (by making the path unique) so it's treated as its own file
+                // This lets us lint for things like namespaces more easily since the namespace in an earlier block
+                // won't be counted towards a later block in the same file
+                $key = dirname($path) . '/' . basename($path, '.md') . "_{$n}" . '.md';
+                $blocks[$key] = [
+                    'content' => $content,
+                    'path' => $key,
+                    'realpath' => $path,
+                    'num' => $n,
+                    'md' => $block,
+                ];
+            }
+        }
+
+        if (PHP_CODESNIFFER_VERBOSITY > 0) {
+            $numBlocks = count($blocks);
+            echo "DONE ($numBlocks codeblocks in queue)" . PHP_EOL;
+        }
+
+        return $blocks;
+    }
+
+    /**
+     * Run the codesniffing rules over the identified markdown codeblocks
+     *
+     * This is very nearly a direct copy of Runner::run()
+     */
+    private function sniff(Runner $sniffer, array $todo): int
+    {
+        // Turn all sniff errors into exceptions.
+        set_error_handler([$sniffer, 'handleErrors']);
+
+        $lastDir  = '';
+        $numBlocks = count($todo);
+
+        // Process each block sequentially - running sniff in parallel isn't supported
+        // We're not actually running this across real files, but we should give the same output we'd get if we were.
+        $numProcessed = 0;
+        /** @var CodeBlock $block */
+        foreach ($todo as $block) {
+            if ($block->ignored === false) {
+                $currDir = dirname($block->realPath);
+                if ($lastDir !== $currDir) {
+                    if (PHP_CODESNIFFER_VERBOSITY > 0) {
+                        echo 'Changing into directory ' . Common::stripBasepath($currDir, $sniffer->config->basepath) . PHP_EOL;
+                    }
+
+                    $lastDir = $currDir;
+                }
+
+                $sniffer->processFile($block);
+            } else if (PHP_CODESNIFFER_VERBOSITY > 0) {
+                echo 'Skipping ' . basename($block->path) . PHP_EOL;
+            }
+
+            $numProcessed++;
+            $sniffer->printProgress($block, $numBlocks, $numProcessed);
+        }
```

Comment:
For this method, everything outside this loop is copied directly from `Runner::run()`, so we probably don't want to change much of that.